### PR TITLE
Upgrade auto-save for surveys and forms

### PR DESCRIPTION
**Pull request by ElviaBth**
🎩 What? Why?
Auto-save for surveys and forms.

📌 Related Issues
#/270 Upgrade 0.28 pending task 2.

📷 Screenshots
image
image

♥️ Thank you!

### CODE REVIEW

File: app/views/decidim/decidim_awesome/admin/config/_constraints.html.erb [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   <%= cell "decidim/announcement", "<p></p>".html_safe, hidden: true %>
 
-  <h5><%= t ".title" %></h5>
+  <h5 class="mb-2 text-base"><%= t ".title" %></h5>
   <% if constraints.blank? %>
     <span class="label warning"><%= t ".always" %></span>
   <% end %>
 
-  <ul class="constraints-list<%= " inactive" if constraints&.detect{|c| c.settings["participatory_space_manifest"]=="none" } %>">
+  <ul class="constraints-list<%= " inactive" if constraints&.detect{|c| c.settings["participatory_space_manifest"]=="none" } %>" style="padding: 0;">
```
Comment (microstudi) [inline app/views/decidim/decidim_awesome/admin/config/_constraints.html.erb:9]:
```suggestion
  <ul class="constraints-list p-0<%= " inactive" if constraints&.detect{|c| c.settings["participatory_space_manifest"]=="none" } %>">
```
We can use tailwind here I think

File: spec/system/admin/admin_spec.rb [hunk 2]
```diff
@@ -169,8 +167,6 @@
   context "when visiting live chat" do
     context "when livechat hacks are enabled" do
       before do
-        skip "Live chat feature is pending to be adapted to Decidim 0.28 and currently is disabled at lib/decidim/decidim_awesome/awesome.rb"
```
Comment (microstudi) [inline spec/system/admin/admin_spec.rb:172]: This PR is not about intergram right? this should stay

File: lib/decidim/decidim_awesome/awesome.rb [hunk 3]
```diff
@@ -46,21 +46,18 @@ module DecidimAwesome
...
     config_accessor :auto_save_forms do
-      # false
-      :disabled
+      true
     end
 
     # Live chat widget linked to Telegram account or group
     # In the admin side only
     config_accessor :intergram_for_admins do
-      # false
```
Comment (microstudi) [inline lib/decidim/decidim_awesome/awesome.rb:56]: this should not be here, this pr is not about intergram

**Review by microstudi — CHANGES_REQUESTED** [review]
@ElviaBth  this PR needs more things

On one hand, it is introducing changes related to the integram chat, this should not be here (see some comments).

On the other hand, this feature is about saving temporary data on forms using the localstorage browser capabilities. This is done with the script "autosave.js"  (which uses https://github.com/appleple/form-storage in a npm package).

You need to:
1. [x] Upgrade the formstorage package (if possible) in package.json
2. [x] Adapt the `autosave.js` file so it removes jQuery dependencies
3. [x] Uncomment the import in the `decidim_application.js` file

Thanks!

File: lib/decidim/decidim_awesome/awesome.rb (hunk 3 above)
Comment (ElviaBth) [reply lib/decidim/decidim_awesome/awesome.rb]: Done

File: spec/system/admin/admin_spec.rb (hunk 2 above)
Comment (ElviaBth) [reply spec/system/admin/admin_spec.rb]: Done

File: app/views/decidim/decidim_awesome/admin/config/_constraints.html.erb (hunk 1 above)
Comment (ElviaBth) [reply app/views/decidim/decidim_awesome/admin/config/_constraints.html.erb:9]: Done

**Pull request merged**